src/main/java/com/ecore/roles/service/impl/MembershipsServiceImpl.java

```diff
@@ -8,48 +9,62 @@
 import com.ecore.roles.repository.MembershipRepository;
 import com.ecore.roles.repository.RoleRepository;
 import com.ecore.roles.service.MembershipsService;
+import com.ecore.roles.service.TeamsService;
 import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
 import lombok.extern.log4j.Log4j2;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import javax.validation.Valid;
 import java.util.List;
+import java.util.Objects;
 import java.util.UUID;
 
-import static java.util.Optional.ofNullable;
-
 @Log4j2
 @Service
+@RequiredArgsConstructor
 public class MembershipsServiceImpl implements MembershipsService {
 
     private final MembershipRepository membershipRepository;
     private final RoleRepository roleRepository;
-
-    @Autowired
-    public MembershipsServiceImpl(
```

Comment:
This constructor can be replaced with @RequiredArgsConstructor lombok annotation in order to become the code cleaner.